Privacy manifest (#6811)
braze-inc/braze-docs: _docs/_developer_guide/platform_integration_guides/swift/privacy_manifest.md

---
nav_title: Apple Privacy Manifest
article_title: Apple Privacy Manifest
page_order: 7
platform: 
  - iOS
description: "This article describes the Braze Privacy Manifest used to declare data collection in your iOS app"
---

# Apple Privacy Manifest

The [Privacy Manifest](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files) is a file published alongside the Braze Swift SDK that describes the data collection practices and "required reasons" APIs.

By default, Braze does not collect any "[tracking data](https://developer.apple.com/app-store/app-privacy-details/#user-tracking)" which Apple describes as data collected and shared with 3rd parties.

The Braze privacy manifest, along with other privacy manifests you collect from other SDKs used, will be aggregated and displayed in the App Store to form your Privacy Nutrition label.

> If you collect other 1st party data in your app, be sure to include that when generating your [Privacy Report](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests#4239187) in Xcode.

Reviewer: ```suggestion
If you collect other 1st party data in your app, be sure to indicate that in your own privacy manifest when generating your [Privacy Report](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests#4239187) in Xcode.
```

## Data Collection

You can review the default data collection policies in our [SDK Data Collection Guide](https://www.braze.com/docs/user_guide/data_and_analytics/user_data_collection/sdk_data_collection/). This list includes attributes the Braze SDK collects to help brands perform better 1st party messaging and segmentation to power meaningful engagement between brands and their users.

Remember, you can always [opt-out](https://www.braze.com/docs/developer_guide/platform_integration_guides/sdk_primer#blocking-data-collection) of the optional data points the Braze SDK collects by default, and your integration determines what additional custom events and attributes you collect.

## Tracking vs. Non-Tracking Data

In addition the default data Braze declares in the Privacy Manifest, you may choose to add additional data to declare as "tracking" prior to publishing your app.

Apple requires that you declare which data types you collect are used for "tracking" vs. "non-tracking" usage. The Braze Swift SDK provides flexible APIs to help you adhere to Apple policies while ensure end-users' privacy is respected.

> To support this privacy enhancement, Braze has added new features to our Swift SDK:

Author: ```suggestion
To support this privacy enhancement, Braze has added new features to our Swift SDK to support customized tracking endpoints and data collection. For a complete tutorial, see our [Braze Swift SDK Tutorial on Privacy Tracking](https://braze-inc.github.io/braze-swift-sdk/tutorials/braze/e1-privacy-tracking/).
```

### New Tracking Domains

> Braze has added new SDK API domains intended for "tracking" data. These domains will prefix `tracking-` to the hostname of your API endpoint and is automatically configured with no integration update needed. For example, if you initialize the SDK with an endpoint of `sdk.iad-01.braze.com`, Braze will send your declared tracking data to `sdk-tracking.iad-01.braze.com`.

Reviewer: ```suggestion
Braze has added new SDK API domains intended for "tracking" data. These domains will embed the `-tracking` suffix within the hostname of your API endpoint and is automatically configured with no integration update needed. For example, if you initialize the SDK with an endpoint of `sdk.iad-01.braze.com`, Braze will send your declared tracking data to `sdk-tracking.iad-01.braze.com`.
```

### Declaring Tracking Data

If you choose to collect user information through the Braze Swift SDK with plans to use this data for [tracking](https://developer.apple.com/app-store/app-privacy-details/#user-tracking) (as described by Apple), you will need to declare this data when submitting your app to the App Store.

Braze provides a [flexible approach](https://braze-inc.github.io/braze-swift-sdk/tutorials/braze/e1-privacy-tracking/) allowing you to dynamically add and remove individual data points from your tracking list.

> For example, to include date of birth, and a few custom events and attributes in your "tracking" list, use the sample code below to reroute these individual data points to the Braze tracking domain. These data points will require a user to have accepted the Ad Tracking Transparency prompt in order to be send succesfully.

Reviewer: ```suggestion
For example, to include date of birth, and a few custom events and attributes in your "tracking" list, use the sample code below to reroute these individual data points to the Braze tracking domain. These data points will require a user to have accepted the Ad Tracking Transparency prompt in order to be send successfully.
```

> 

Author: ```suggestion

For the full list of data types that can be configured for tracking, refer to the tracking properties [here](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/configuration-swift.class/trackingproperty).
```

---
nav_title: Apple Privacy Manifest
article_title: Apple Privacy Manifest
page_order: 7
platform: 
  - iOS
description: "This article describes the Braze Privacy Manifest used to declare data collection in your iOS app"
---

# Apple Privacy Manifest

The [Privacy Manifest](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files) is a file published alongside the Braze Swift SDK that describes the data collection practices and "required reasons" APIs.

By default, Braze does not collect any "[tracking data](https://developer.apple.com/app-store/app-privacy-details/#user-tracking)" which Apple describes as data collected and shared with 3rd parties.

> The Braze privacy manifest, along with other privacy manifests you collect from other SDKs used, will be aggregated and displayed in the App Store to form your Privacy Nutrition label.

Reviewer 2: ```suggestion
The Braze Privacy Manifest, along with other privacy manifests you collect from other SDKs used, will be aggregated and displayed in the App Store to form your Privacy Nutrition label.
```

Overall I believe the style should be
- capitalize the Braze Privacy Manifest product
- lowercase privacy manifest as a general or Apple concept (as Apple does in their own docs, e.g. https://developer.apple.com/documentation/bundleresources/privacy_manifest_files)

Reviewer 3: Thanks for the note here! We are going to keep this lowercase for now to be in-line with Apple's documentation. Let me know if you have any questions!

> By default, the Braze SDK does not collect tracking data. However, depending on your Braze SDK configuration, you may be required to add a privacy manifest to your app.

Author: ```suggestion
By default, the Braze SDK does not collect tracking data. However, depending on your Braze SDK configuration, you may be required to specify this data in your app's privacy manifest.
```

@jerielng does that sound right?

Reviewer: Yep that's correct!

## What is a privacy manifest?

A privacy manifest is a file in your Xcode project that describes the reason your app and third-party SDKs collect data, along with the data-collection method. Each of your third-party SDKs that track data require its own privacy manifest file. When you [create your app's privacy report](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests#4239187), these privacy manifest files are automatically aggregated into a single report.

> When an end-user launches your app for the first time, your app's privacy report will be used to generate an Ad Tracking Transparency agreement. Starting with iOS 17.2, Apple will block all tracking data in your app until the end-user accepts this agreement.

Author: ```suggestion
Starting with iOS 17.2, Apple will block all declared tracking endpoints in your app until the end-user accepts the Ad Tracking Transparency prompt. Braze has added a new tracking endpoint that will automatically route declared tracking data, while keeping all non-tracking 1st party data to the original endpoint.
```

## Declaring Braze tracking data

### Step 1: Review your current policies

> Review your Braze SDK's current data-collection policies with your legal team to determine whether your app collects tracking data [as defined by Apple](#what-is-tracking-data). If you're not collecting any tracking data, you don't need to add a privacy manifest at this time. For more information about the Braze SDK's data-collection policies, see [SDK data collection]({{site.baseurl}}/user_guide/data_and_analytics/user_data_collection/sdk_data_collection/).

Author: ```suggestion
Review your Braze SDK's current data-collection policies with your legal team to determine whether your app collects tracking data [as defined by Apple](#what-is-tracking-data). If you're not collecting any tracking data, you don't need to customize your privacy manifest for Braze at this time. For more information about the Braze SDK's data-collection policies, see [SDK data collection]({{site.baseurl}}/user_guide/data_and_analytics/user_data_collection/sdk_data_collection/).
```

Reviewer: Yep this suggestion is more precise. App developers will still need to determine whether they need their own privacy manifest for reasons other than collecting tracking data, such as using one of the [required reasons APIs](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api).

{% alert important %}
If any of your non-Braze SDKs collect tracking data, you'll need to review those policies separately.
{% endalert %}

### Step 2: Add the Braze privacy manifest

> [Download the Braze privacy manifest](https://github.com/braze-inc/braze-swift-sdk/blob/main/Sources/BrazeKitResources/Resources/PrivacyInfo.xcprivacy) from GitHub, then open your Xcode project and move the file into `Brazekit.bundle`.

Author: @jerielng is this correct?

Reviewer: It's already included automatically in most of our integrations (SPM, CocoaPods). The bundle is only used for manual integrations of static XCFrameworks because the resources need to be linked manually, which is already called out in the manual integration [docs](https://www.braze.com/docs/developer_guide/platform_integration_guides/swift/initial_sdk_setup/installation_methods/manual_integration/), so I don't think we need this section.

Reviewer: If we want to, we could also call out which versions of the Swift SDK they will need to use (7.5.0+ for just the privacy manifest files, 7.6.0+ for the customization APIs).

### Step 3: Declare your tracking data

{% alert tip %}
For a full walkthrough, see [Tutorial: Privacy Tracking Data](https://braze-inc.github.io/braze-swift-sdk/tutorials/braze/e1-privacy-tracking/).
{% endalert %}

Open `AppDelegate.swift`, then list each [tracking property](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/configuration-swift.class/trackingproperty/) you want to declare. Your declaration should be similar to the following, where `dateOfBirth`, `customEvent`, and `customAttribute` represent declared tracking properties. 

> 

Author: should add a note that once this data is added, it will be blocked until the user accepts the ATT prompt. So they should only add these fields when they consider them tracking.

For the full list of data types that can be configured for tracking, refer to the tracking properties [here](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/configuration-swift.class/trackingproperty).

> ```swift

Reviewer 2: ```suggestion

```swift
```

import BrazeKit

@main
class AppDelegate: UIResponder, UIApplicationDelegate {

  static var braze: Braze? = nil

  func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
    let configuration = Braze.Configuration(apiKey: brazeApiKey, endpoint: brazeEndpoint)
    // Declare which types of data you wish to collect for user tracking.
    configuration.api.trackingPropertyAllowList = [
      .dateOfBirth,
      .customEvent(["event-1"]),
      .customAttribute(["attribute-1", "attribute-2"])
    ]
    let braze = Braze(configuration: configuration)
    AppDelegate.braze = braze
    return true
  }
}
```

> Next, configure your tracking list, so data points are added and removed automatically. In the following example, the tracking list is set to automatically update after a user accepts the Ad Tracking Transparency agreement.

Author: more like an optional alternative approach, to update/set it mid session
